ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py

```diff
@@ -1318,3 +1318,123 @@ def test_torch_clone(
         on_device=on_device,
         input=x[0],
     )
+
+
+@st.composite
+def _get_dtype_value1_value2_cov(
+    draw,
+    available_dtypes,
+    min_num_dims,
+    max_num_dims,
+    min_dim_size,
+    max_dim_size,
+    abs_smallest_val=None,
+    min_value=None,
+    max_value=None,
+    allow_inf=False,
+    exclude_min=False,
+    exclude_max=False,
+    large_abs_safety_factor=4,
+    small_abs_safety_factor=4,
+    safety_factor_scale="log",
+):
+    shape = draw(
+        helpers.get_shape(
+            allow_none=False,
+            min_num_dims=min_num_dims,
+            max_num_dims=max_num_dims,
+            min_dim_size=min_dim_size,
+            max_dim_size=max_dim_size,
+        )
+    )
+
+    dtype = draw(st.sampled_from(draw(available_dtypes)))
+
+    values = []
+    for i in range(1):
+        values.append(
+            draw(
+                helpers.array_values(
+                    dtype=dtype,
+                    shape=shape,
+                    abs_smallest_val=abs_smallest_val,
+                    min_value=min_value,
+                    max_value=max_value,
+                    allow_inf=allow_inf,
+                    exclude_min=exclude_min,
+                    exclude_max=exclude_max,
+                    large_abs_safety_factor=large_abs_safety_factor,
+                    small_abs_safety_factor=small_abs_safety_factor,
+                    safety_factor_scale=safety_factor_scale,
+                )
+            )
+        )
+
+    value1 = values[0]
+
+    correction = draw(helpers.ints(min_value=0, max_value=1))
+
+    fweights = draw(
+        helpers.array_values(
+            dtype="int64",
+            shape=shape[1],
+            abs_smallest_val=1,
+            min_value=1,
+            max_value=10,
+            allow_inf=False,
+        )
+    )
+
+    aweights = draw(
+        helpers.array_values(
+            dtype="float64",
+            shape=shape[1],
+            abs_smallest_val=1,
+            min_value=1,
+            max_value=10,
+            allow_inf=False,
+            small_abs_safety_factor=1,
+        )
+    )
+
+    return [dtype], value1, correction, fweights, aweights
+
+
+# cov
+@handle_frontend_test(
+    fn_tree="torch.cov",
+    dtype_x1_corr_cov=_get_dtype_value1_value2_cov(
+        available_dtypes=helpers.get_dtypes("float"),
+        min_num_dims=2,
+        max_num_dims=2,
+        min_dim_size=2,
+        max_dim_size=5,
+        min_value=1,
+        max_value=1e10,
+        abs_smallest_val=0.01,
+        large_abs_safety_factor=2,
+        safety_factor_scale="log",
+    ),
+    test_with_out=st.just(False),
+)
+def test_torch_cov(
+    dtype_x1_corr_cov,
+    test_flags,
+    frontend,
+    fn_tree,
+    on_device,
+):
+    dtype, x1, correction, fweights, aweights = dtype_x1_corr_cov
+    helpers.test_frontend_function(
+        input_dtypes=["float64", "int64", "float64"],

```

Comment:
These should not be restricted, I also requested for similar changes on your other PR too :slightly_smiling_face: